src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +109,447 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+    posRot.z = this->actor.posRot.pos.z;
+
+    func_80033480(globalCtx, &posRot, scaleX1, 2, scaleX2, 0xA0, 1);
+}
+
+void func_808B5240(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    f32 tempUnk6;
+    f32 tempUnk2;
+    s16 index;
+    Vec3f position;
+    Vec3f* actorPosition = &this->actor.posRot.pos;
+
+    if (true) {}
+
+    while (true) {
+        if (this->unk_158 >= 26U || this->unk_154 < D_808B5EB0[this->unk_158].unk_0) {
+            break;
+        }
+
+        index = this->unk_158;
+
+        tempUnk2 = D_808B5EB0[index].unk_2;
+        tempUnk6 = D_808B5EB0[index].unk_6;
+
+        position.x = ((this->sinRotation * tempUnk6) + (tempUnk2 * this->cosRotation)) + actorPosition->x;
+        position.y = D_808B5EB0[index].unk_4 + actorPosition->y;
+        position.z = ((this->cosRotation * tempUnk6) - (tempUnk2 * this->sinRotation)) + actorPosition->z;
+
+        func_800287AC(globalCtx, &position, &velocity, &acceleration, D_808B5EB0[index].scale,
+                      D_808B5EB0[index].scaleStep, D_808B5EB0[index].life);
+
+        this->unk_158 += 1;
+    }
+}
+
+void func_808B53A8(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    f32 velocityYMultiplier = 1.3f;
+    Vec3f randomPosition;
+    Vec3f randomVelocity;
+    s32 index;
+    s16 scale;
+
+    if (this->actor.params == 0) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_SPOT16_BOMBSTONE, this->actor.posRot.pos.x,
+                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 5);
+        index = 3;
+    } else {
+        index = 0;
+    }
+
+    if (index < ARRAY_COUNT(D_808B6074)) {
+        do {
+            randomPosition.x = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.x;
+            randomPosition.y = ((Math_Rand_ZeroOne() * 5.0f) + this->actor.posRot.pos.y) + 8.0f;
+            randomPosition.z = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.z;
+
+            randomVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+            randomVelocity.y = (Math_Rand_ZeroOne() * 14.0) + (fabsf(this->actor.velocity.y) * velocityYMultiplier);
+            randomVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+
+            scale = D_808B6074[index] * this->actor.scale.x * 3;
+
+            func_80029E8C(globalCtx, &randomPosition, &randomVelocity, &this->actor.posRot.pos, -0x1A4, 0x31, 0xF, 0xF,
+                          0, scale, 2, 0x40, 0xA0, -1, OBJECT_BOMBIWA, &D_060009E0);
+            index += 1;
+        } while (index != ARRAY_COUNT(D_808B6074));
+    }
+}
+
+void func_808B561C(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    s32 index;
+    PosRot* posRot;
+
+    posRot = &this->actor.posRot;
+    for (index = 0; index < ARRAY_COUNT(D_808B6088); index++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xCD, posRot->pos.x, posRot->pos.y, posRot->pos.z, 0, 0, 0,
+                        D_808B6088[index]) == 0) {
```

Comment:
```suggestion
        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_SPOT16_BOMBSTONE, posRot->pos.x, posRot->pos.y, posRot->pos.z, 0, 0, 0,
                        D_808B6088[index]) == NULL) {
```
use NULL to compare pointers